### PR TITLE
MGMT-7693: More V2 subsystem tests

### PR DESCRIPTION
**Pull request by slaviered**
# Assisted Pull Request

## Description
Add Bind/Unbind flows to REST API subsystem tests for host and cluster

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: subsystem/cluster_v2_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+				SSHAuthorizedKey: swag.String(sshPublicKey),
+				ImageType:        models.ImageTypeFullIso,
+			},
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		infraEnv = infraReq.GetPayload()
+	})
+
+	JustBeforeEach(func() {
```
Comment (carbonin) [inline subsystem/cluster_v2_test.go:56]: Why separate this out?
I don't really think this is the intended use of `JustBeforeEach`
ref: https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach

**Review by carbonin — COMMENTED** [review]
Please don't ignore the PR template.

Also, can you link to the PR that added this functionality (apparently without tests)?

**Comment by slaviered** [comment]
/test e2e-metal-assisted-operator-ztp



<br class="Apple-interchange-newline">

File: subsystem/cluster_v2_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	var infraEnvID strfmt.UUID
+	var ips []string
+	var h1, h2, h3 *models.Host
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	BeforeEach(func() {
+		clusterReq, err := userBMClient.Installer.RegisterCluster(ctx, &installer.RegisterClusterParams{
```
Comment (rollandf) [inline subsystem/cluster_v2_test.go:28]: We should use V2 API here also

Comment (rollandf) [reply subsystem/cluster_v2_test.go:28]: /hold
Need to use V2 API when creating the cluster

Comment (slaviered) [reply subsystem/cluster_v2_test.go:28]: fixed

**Pull request merged**